### PR TITLE
Cleaning QOL

### PR DESCRIPTION
**Pull request by flappybatpal**
[QOL][BUG]
## About the PR 

1. Adds a clean_stains() proc to clothing that removes all stains.
2. Makes clean_forensic() call clean_stains() on clothing.
3. Directly cleans items that aren't on a turf (i.e. in contents) when sprayed by a spray bottle.
4. Fixes mop attack intent check using the wrong intent var.
5. Prevent pointless "Mop is dry" messages when a  dry mop is used on a person.
6. Adds the attack intent check to sponges so you can wipe someone down without "attacking" them.
7. Reduce sponge stamina damage from default (20) to 5. It's a sponge!
8. Fix sponge menu options building a list using += which would add hidden duplicate menu options and prevent the intended automatic selection when there is only a single action available.
9. Moves an inline return for code readability.

## Why's this needed? 
Cleaning did not change the name of items, so they remained "blood soaked".
Using the spray bottle on items in your pocket/belt/bag or in a container inventory would not clean them.
Mops and sponges often get clicked on people moving around. Changing these to no longer whack people with red text reduces confusion.

### CODE REVIEW

File: code/modules/forensics/atom_forensic.dm [hunk 1]
```diff
@@ -201,6 +199,9 @@
 				#else
 				CI.UpdateOverlays(null, "blood_splatter")
 				#endif
+				if (istype(src, /obj/item/clothing))
+					var/obj/item/clothing/C = src
+					C.clean_stains()
```
Comment (Tarmunora) [inline code/modules/forensics/atom_forensic.dm:204]: does this need to be at this indentation level?

File: code/modules/forensics/atom_forensic.dm [hunk 2]
```diff
@@ -238,6 +239,9 @@
 						#else
 						check.UpdateOverlays(null, "blood_splatter")
 						#endif
+						if (istype(check, /obj/item/clothing))
+							var/obj/item/clothing/C = check
+							C.clean_stains()
```
Comment (Tarmunora) [inline code/modules/forensics/atom_forensic.dm:244]: ditto

File: code/obj/item/mops_cleaners.dm [hunk 3]
```diff
@@ -208,14 +208,17 @@ WET FLOOR SIGN
...
 		JOB_XP(user, "Janitor", 2)
-
 	var/obj/decal/D = new/obj/decal(get_turf(src))
 	D.name = "chemicals"
 	D.icon = 'icons/obj/chemical.dmi'
 	D.icon_state = "chempuff"
 	D.create_reagents(5) // cogwerks: lowered from 10 to 5
 	src.reagents.trans_to(D, 5)
 	playsound(src.loc, "sound/effects/zzzt.ogg", 50, 1, -6)
+	// Direct cleaning for objects or items directly sprayed in containers are not cleaned
```
Comment (Tarmunora) [inline code/obj/item/mops_cleaners.dm:218]: word salad?

File: code/obj/item/mops_cleaners.dm [hunk 4]
```diff
@@ -208,14 +208,17 @@ WET FLOOR SIGN
...
+	if (istype(A, /obj/item))
+		D.reagents.reaction(A)
+		D.reagents.remove_any(1)
```
Comment (Tarmunora) [inline code/obj/item/mops_cleaners.dm:221]: Might want to rework the proc so that the ranged spray doesn't happen if spraying something directly

**Review by Tarmunora — CHANGES_REQUESTED** [review]

File: code/obj/item/mops_cleaners.dm (hunk 4 above)
Comment (flappybatpal) [reply code/obj/item/mops_cleaners.dm:221]: Possibly check if the item is in contents? It seems like the overall intention for the spray is to clean multiple tiles and items in one go.

File: code/modules/forensics/atom_forensic.dm (hunk 1 above)
Comment (flappybatpal) [reply code/modules/forensics/atom_forensic.dm]: Moved up.

File: code/modules/forensics/atom_forensic.dm (hunk 2 above)
Comment (flappybatpal) [reply code/modules/forensics/atom_forensic.dm]: Moved up.

File: code/obj/item/mops_cleaners.dm (hunk 4 above)
Comment (flappybatpal) [reply code/obj/item/mops_cleaners.dm]: Reworked this to do a turf check.

File: code/obj/item/mops_cleaners.dm [hunk 5]
```diff
@@ -208,14 +208,19 @@ WET FLOOR SIGN
 
 	if(src.reagents.has_reagent("water") || src.reagents.has_reagent("cleaner"))
 		JOB_XP(user, "Janitor", 2)
-
+	playsound(src.loc, "sound/effects/zzzt.ogg", 50, 1, -6)
+	// Make sure we clean an item that was sprayed directly in case it is in contents
+	if (!isturf(A.loc))
+		if (istype(A, /obj/item))
```
Comment (flappybatpal) [inline code/obj/item/mops_cleaners.dm:214]: I'm not sure what could be sprayed on a turf that isn't on a turf and wouldn't be an /obj/item

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**